### DISCUSSION TITLE
Rewind is not supported in Isolated mode

### DISCUSSION THREAD

**Discussion by Michson07** [microsoft/DurableFunctionsMonitor#197] [Q&A]
Hi!
I migrated azure durable functions from inProcess to Isolated mode recently.
I used "Rewind" option several times and I just noticed that this option is no longer supported in isolated mode (I am using version 6.5.0-beta2).
What was the decision behind it? Is it going to be implemented? 
It's quite important feature, especially when the bug is fixed and you want to continue an existing instance.
![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/55213234/e1321c9f-cb9d-4622-ad7b-c809c35214d4)

**Comment by scale-tone** [comment]
@Michson07 , ["Rewind" is a feature that has always been in preview](https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-http-api#rewind-instance-preview). 

In Isolated mode the DurableTaskClient [simply does not have such a method](https://learn.microsoft.com/en-us/dotnet/api/microsoft.durabletask.client.durabletaskclient?view=durabletask-dotnet-1.x), so there's no way for DfMon to support it.

To check the work state on that feature, I believe, you can follow [this thread on the DurableTask repo](https://github.com/Azure/durabletask/issues/731).

**Comment by michal-czajkowski** [comment]
@scale-tone do you have any updates on this? I see there’s no reply in the discussion you mentioned, but perhaps you’ve found a workaround.

**Reply by epDugas** [reply]
Just to surface, rewind was "recently" added in isolated (https://github.com/microsoft/durabletask-dotnet/releases/tag/v1.17.0). @scale-tone - if I get your blessing to update the dependency, I will attempt to implement.

**Reply by scale-tone** [reply]
That would be really great, @epDugas ! Thanks again!

**Reply by Michson07** [reply]
@epDugas When are you planning to update this? Let me know if you need any help!

**Reply by epDugas** [reply]
Apologies for the delayed response. Commented on @Michson07 #315.

**Comment by Michson07** [comment]
@scale-tone I’ve created a pull request with an improvement. When you have a moment, could you please take a look?
If the change looks good, it would be great if a new package version could be published as well.
Thanks a lot for your time and work on this project!

#315

**Reply by scale-tone** [reply]
Thanks so much @Michson07 , great stuff! 
Merged. Expect a new package in the next few days.

**Reply by Michson07** [reply]
@scale-tone  Great, thanks! Do you know approximately when the new version will be released?

**Reply by scale-tone** [reply]
Looks like bumping up Microsoft.Azure.Functions.Worker.Extensions.DurableTask also requires bumping up Microsoft.Azure.Functions.Worker to 2.51.0 (as @epDugas correctly pointed out in the PR comments).

Otherwise I'm getting strange errors:
```
[2026-04-17T14:19:08.674Z] Executed 'Functions.DfmServeStaticsFunction' (Failed, Id=e507f531-06e4-4d01-821a-21bab9010561, Duration=201ms)
[2026-04-17T14:19:08.683Z] System.Private.CoreLib: Exception while executing function: Functions.DfmServeStaticsFunction. System.Private.CoreLib: Result: Failure
Exception: Method not found: 'Void Microsoft.Azure.Functions.Worker.DefaultTraceContext..ctor(System.String, System.String)'.
Stack:    at Microsoft.Azure.Functions.Worker.Grpc.GrpcFunctionInvocation..ctor(InvocationRequest invocationRequest)
[2026-04-17T14:19:08.692Z]    at Microsoft.Azure.Functions.Worker.Handlers.InvocationHandler.InvokeAsync(InvocationRequest request) in /_/src/DotNetWorker.Grpc/Handlers/InvocationHandler.cs:line 75.
```

Will bump them up shortly...

**Reply by scale-tone** [reply]
Pushed [v6.8.0-beta1](https://www.nuget.org/packages/DurableFunctionsMonitor.DotNetIsolated/6.8.0-beta1). Can you folks check if anything's missing?

**Reply by epDugas** [reply]
Package bumps look good. I pulled the beta, and attempted a Rewind, and the operation submission worked, but unfortunately I don't have any orchestrations to verify on, because rewind doesn't support orchestrations with Durable Timers :(.

**Reply by Michson07** [reply]
> Pushed [v6.8.0-beta1](https://www.nuget.org/packages/DurableFunctionsMonitor.DotNetIsolated/6.8.0-beta1). Can you folks check if anything's missing?

I just checked and Rewind works perfectly

**Comment by scale-tone** [comment]
Released [v6.8.1](https://github.com/microsoft/DurableFunctionsMonitor/releases/tag/v6.8.1) with support for this.
Thanks again, everyone!
